File: leetcode/go/204.count-primes.go
import (
    "math"
)
func countPrimes(n int) int {
    if n < 3 {
        return 0;
    }
    vec := make([]bool, n);
    for i,_ := range vec {vec[i] = true}
    vec[0] = false
    vec[1] = false
    sqrtn := int(math.Sqrt(float64(n)))
    i := 2
    for i <= sqrtn {
        if vec[i] {
            for j:=2 * i; j < n; j += i {
                vec[j] = false
            }
        }
        i++
    }
    count := 0
    for _, b := range vec {
        if b {
            count++
        }
    }
    return count
}

File: leetcode/rust-leetcode/src/204.count-primes.rs
impl Solution {
    pub fn count_primes(n: i32) -> i32 {
        if n < 3 {
            return 0;
        }
        let n = (n as usize);
        let mut vec = vec![true; n];
        vec[0] = false;
        vec[1] = false;
        let sqrtn = f64::sqrt(n as f64) as usize;
        let mut count = 0;
        for i in 2..=sqrtn {
            if vec[i] {
                for j in (2*i..n).step_by(i) {
                    vec[j] = false;
                }
                count += 1;
            }
        }
        vec.into_iter().filter(|&b| b).count() as _
    }
}

